src/core/pyproxy.c

```diff
@@ -1225,9 +1232,69 @@ EM_JS_NUM(int, pyproxy_init_js, (), {
 });
 // clang-format on
 
+JsRef
+create_once_proxy(PyObject* obj)
+{
+  Py_INCREF(obj);
+  return (JsRef)EM_ASM_INT(
+    {
+      let o = { $$ : { ptr : $0 } };
+      function wrapper(... args)
+      {
+        if (!o) {
+          throw new Error("OnceProxy can only be called once");
+        }
+        try {
+          return Module.PyProxyClass.prototype.apply.call(o, undefined, args);
+        } finally {
+          o = undefined;
+          _Py_DecRef($0);
+        }
+      }
+      return Module.hiwire.new_value(wrapper);
+    },
+    obj);
+}
+
+static PyObject*
+create_once_proxy_py(PyObject* _mod, PyObject* obj)
+{
+  JsRef ref = create_once_proxy(obj);
+  PyObject* result = JsProxy_create(ref);
+  hiwire_decref(ref);
+  return result;
+}
+
+static PyObject*
+create_proxy(PyObject* _mod, PyObject* obj)
+{
+  JsRef ref = pyproxy_new(obj);
+  PyObject* result = JsProxy_create(ref);
+  hiwire_decref(ref);
+  return result;
+}
+
+static PyMethodDef pyproxy_methods[] = {
+  {
+    "create_once_proxy",
+    create_once_proxy_py,
+    METH_O,
+    PyDoc_STR("Create a wrapper around a Python function that can be called "
+              "once from Javascript"),
```

Comment:
I think the first line of the docstring should be a signature line here and below?
https://www.python.org/dev/peps/pep-0007/#documentation-strings